decentraland/unity-explorer
# Fix: sdk7 text shape alignment

## What does this PR change?
fix #879, 
fix #881

Main problem was due to the re-used values from the pooled TMPro object transforms, that were never reset.
- replicate most of the unity-renderer logic related to the protobuf values
- replaced logic by the usage of generic pool
- moved some logic into Systems and Extensions
- reused MaterialPropertyBlock 

## How to test the changes?

1. Launch the explorer
2. Goto dclexperience
3. Verify that text is on proper places. For example in Carousel - JUMP-IN, dates, titles
NOTE:  while most of texts will be on the proper place, you will also see text that should be hidden - this will be fixed in another PR
4. Verify you don't see a lot of black/white textures at the bar at 0,0 (same position as in #708)

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
